tiledb/sm/storage_manager/consolidator.cc

```diff
@@ -87,135 +309,131 @@ Status Consolidator::consolidate(
-    return Status::Ok();
-  }
-
   // Read from one array and write to the other
   st = copy_array(query_r, query_w);
   if (!st.ok()) {
-    storage_manager_->array_close(array_uri, QueryType::READ);
-    storage_manager_->array_close(array_uri, QueryType::WRITE);
-    clean_up(subarray, buffer_num, buffers, buffer_sizes, query_r, query_w);
+    storage_manager_->array_close_for_reads(array_uri);
+    storage_manager_->array_close_for_writes(array_uri);
+    clean_up(buffer_num, buffers, buffer_sizes, query_r, query_w);
     return st;
   }
 
-  // Get old fragment uris
-  old_fragment_uris = query_r->fragment_uris();
-
   // Close array for reading
-  st = storage_manager_->array_close(array_uri, QueryType::READ);
+  st = storage_manager_->array_close_for_reads(array_uri);
   if (!st.ok()) {
-    storage_manager_->array_close(array_uri, QueryType::WRITE);
-    storage_manager_->vfs()->remove_dir(new_fragment_uri);
-    clean_up(subarray, buffer_num, buffers, buffer_sizes, query_r, query_w);
+    storage_manager_->array_close_for_writes(array_uri);
+    storage_manager_->vfs()->remove_dir(*new_fragment_uri);
+    clean_up(buffer_num, buffers, buffer_sizes, query_r, query_w);
     return st;
   }
 
   // Lock the array exclusively
   st = storage_manager_->array_xlock(array_uri);
   if (!st.ok()) {
-    storage_manager_->array_close(array_uri, QueryType::WRITE);
-    storage_manager_->vfs()->remove_dir(new_fragment_uri);
-    clean_up(subarray, buffer_num, buffers, buffer_sizes, query_r, query_w);
+    storage_manager_->array_close_for_writes(array_uri);
+    storage_manager_->vfs()->remove_dir(*new_fragment_uri);
+    clean_up(buffer_num, buffers, buffer_sizes, query_r, query_w);
     return st;
   }
 
-  // Finalize both queries
+  // Finalize write query
   st = query_w->finalize();
   if (!st.ok()) {
-    storage_manager_->array_close(array_uri, QueryType::WRITE);
-    clean_up(subarray, buffer_num, buffers, buffer_sizes, query_r, query_w);
+    storage_manager_->array_close_for_writes(array_uri);
+    clean_up(buffer_num, buffers, buffer_sizes, query_r, query_w);
     storage_manager_->array_xunlock(array_uri);
-    storage_manager_->vfs()->remove_dir(new_fragment_uri);
+    bool is_dir;
+    auto st2 = storage_manager_->vfs()->is_dir(*new_fragment_uri, &is_dir);
+    if (is_dir)
+      storage_manager_->vfs()->remove_dir(*new_fragment_uri);
     return st;
   }
 
   // Close array
-  st = storage_manager_->array_close(array_uri, QueryType::WRITE);
+  storage_manager_->array_close_for_writes(array_uri);
   if (!st.ok()) {
     storage_manager_->array_xunlock(array_uri);
-    clean_up(subarray, buffer_num, buffers, buffer_sizes, query_r, query_w);
-    storage_manager_->vfs()->remove_dir(new_fragment_uri);
+    clean_up(buffer_num, buffers, buffer_sizes, query_r, query_w);
+    bool is_dir;
+    auto st2 = storage_manager_->vfs()->is_dir(*new_fragment_uri, &is_dir);
```

Comment:
`st2` not checked

Reply:
These two places handle a previous error status. Since we do not provide an error stack (yet), checking `st2` here is redundant, since we will report only `st`. However, we need to initialize `is_dir` to `false` to properly handle the `if` command immediately below.